src/decompose.rs

```diff
@@ -69,63 +74,72 @@ impl<I: Iterator<Item=char>> Iterator for Decompositions<I> {
     fn next(&mut self) -> Option<char> {
         use self::DecompositionType::*;
 
-        match self.buffer.first() {
-            Some(&(c, 0)) => {
-                self.sorted = false;
-                self.buffer.remove(0);
```

Comment:
oh interesting, this was doing a reallocation on every emitted character!  nice work switching to `VecDeque`